File: algorithms/1957.delete-characters-to-make-fancy-string/solution.go
func makeFancyString(s string) string {
	res := []rune{}

	last := ' '
	second_last := ' '

	for _, r := range s {
		if r != last || r != second_last {
			res = append(res, r)
		}

		second_last = last
		last = r
	}

	return string(res)
}

File: algorithms/1957.delete-characters-to-make-fancy-string/solution.rs
impl Solution {
    pub fn make_fancy_string(s: String) -> String {
        let mut res = vec![];

        let mut last = ' ';
        let mut second_last = ' ';

        for c in s.chars() {
            if c != last || c != second_last {
                res.push(c);
            }

            second_last = last;
            last = c;
        }

        res.iter().collect::<String>()
    }
}
